draft-ietf-jsonpath-base.md

```diff
@@ -902,11 +965,173 @@ size doesn't
+
+### Union Selector
+
+#### Syntax
+
+The union selector is syntactically related to the `index-selector`. It contains multiple, comma separated entries.
 
+~~~~ abnf
+union-selector = "[" ws union-entry 1*(ws "," ws union-entry) ws "]"
 
+union-entry    =  ( quoted-member-name /
+                    element-index      /
+                    slice-index
+                  )
+~~~~
+
+> Task (T1): This, besides slice-index, is currently one of only two
+> places in the document that mentions whitespace.  Whitespace needs
+> to be handled throughout the ABNF syntax.  Room Consensus at the
+> 2021-06-15 interim was that JSONPath generally is generous with
+> allowing insignificant whitespace throughout.  Minimizing the impact
+> of the many whitespace insertion points by choosing a rule name such
+> as "S" was mentioned.  Some conventions will probably help with
+> minimizing the number of places where S needs to be inserted.
+
+#### Semantics
 
+A union selects any node which is selected by at least one of the union selectors and selects the concatenation of the
+lists (in the order of the selectors) of nodes selected by the union elements.
+Note that any node selected in more than one of the union selectors is kept
+as many times in the node list.
 
 
+### Filter Selector
+
+#### Syntax
+
+The filter selector has the form `[?<expr>]`. It works via iterating over structured values, i.e. arrays and objects.
+
+~~~~ abnf
+filter-selector    = "[?" boolean-expr "]"
+~~~~
+
+During iteration process each array element or object member is visited and its value — accessible via symbol `@` — or one of its descendants — uniquely defined by a relative path — is tested against a boolean expression `boolean-expr`.
+
+The current item is selected if and only if the result is `true`.
+
+
+~~~~ abnf
+
+
+boolean-expr = logical-expr
+logical-expr = ([neg-op] primary-expr) / logical-or-expr
+neg-op       = "!"                                  ; not operator
+primary-expr = "(" logical-or-expr ")"
+logical-or-expr = logical-and-expr *["||" logical-and-expr]
+logical-and-expr = comp-expr *["&&" comp-expr]
+
+comp-expr    = (rel-path-val) [(comp-op comparable /  ; comparison
+                            regex-op regex     /  ; RegEx test
+                            in-op container )]    ; containment test
+comp-op      = "==" / "!=" /                        ; comparison ...
+               "<"  / ">"  /                        ; operators
+               "<=" / ">="
+regex-op     = "~="                                 ; RegEx match
+in-op        = " in "                               ; in operator
+comparable   = number / string-literal /            ; primitive ...
+               true / false / null /                ; values only
+               rel-path-val /                       ; descendant value
+               json-path                            ; any value
+
+rel-path-val = "@" *(dot-selector / index-selector)
```

Comment:
We also need absolute path values such as `$.foo`.

Reply:
Todo for next PR...

Reply:
this is a `json-path` then.

Reply:
That is a simple solution that we can refine later.  Put it in.